packages/core/src/components/fields/BooleanField.tsx

```diff
@@ -22,17 +32,24 @@ function BooleanField(props) {
   } = props;
   const { title } = schema;
   const { widgets, formContext } = registry;
-  const { widget = "checkbox", ...options } = getUiOptions(uiSchema);
+  const { widget = "checkbox", ...options } = getUiOptions<T, F>(uiSchema);
   const Widget = getWidget(schema, widget, widgets);
 
   let enumOptions;
 
   if (Array.isArray(schema.oneOf)) {
     enumOptions = optionsList({
-      oneOf: schema.oneOf.map((option) => ({
-        ...option,
-        title: option.title || (option.const === true ? "Yes" : "No"),
-      })),
+      oneOf: schema.oneOf
+        .map((option: RJSFSchemaDefinition) => {
+          if (isObject(option)) {
+            return {
+              ...option,
+              title: option.title || (option.const === true ? "Yes" : "No"),
+            };
+          }
+          return undefined;
+        })
+        .filter((o) => o) as RJSFSchemaDefinition[], // cast away the error that typescript can't grok is fixed
```

Comment:
```suggestion
        .filter((o): o is RJSFSchemaDefinition => o) as // cast away the error that typescript can't grok is fixed
```

Reply:
Hmmm, not happy with that:
<img width="911" alt="Screen Shot 2022-08-12 at 9 17 17 AM" src="https://user-images.githubusercontent.com/51679588/184399471-d04bbc06-dec6-479f-9f96-95e23afdc29b.png">